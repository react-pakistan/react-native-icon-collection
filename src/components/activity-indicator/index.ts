export {
  ActivityIndicator,
  ActivityIndicatorSize,
} from './activity-indicator';
