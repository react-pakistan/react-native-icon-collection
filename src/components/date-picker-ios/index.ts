export { DatePickerIOS } from './date-picker-ios';
