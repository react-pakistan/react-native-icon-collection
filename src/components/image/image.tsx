import styled from 'styled-components/native';

export const Image = styled.Image`
  height: ${({ height } : IImageProps) : number => height || 100}px;
  width: ${({ width } : IImageProps) : number => width || 100}px;
`;

export interface IImageProps {
  /**
   *
   */
  height : number;
  /**
   *
   */
  width : number;
}

export enum ResizeMode {
  center = 'center',
  contain = 'contain',
  cover = 'cover',
  repeat = 'repeat',
  stretch = 'stretch',
}

export enum ResizeMethod {
  auto = 'auto',
  resize = 'resize',
  scale = 'scale',
}
