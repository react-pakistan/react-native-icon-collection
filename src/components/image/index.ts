export {
  IImageProps,
  Image,
  ResizeMethod,
  ResizeMode,
} from './image';
