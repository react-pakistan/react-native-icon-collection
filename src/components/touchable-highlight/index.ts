export { TouchableHighlight } from './touchable-highlight';
