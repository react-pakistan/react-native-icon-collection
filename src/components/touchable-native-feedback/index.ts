export { TouchableNativeFeedback } from './touchable-native-feedback';
