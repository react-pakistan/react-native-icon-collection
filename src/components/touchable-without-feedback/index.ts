export { TouchableWithoutFeedback } from './touchable-without-feedback';
