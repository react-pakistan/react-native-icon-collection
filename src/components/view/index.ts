export { View, IView } from './view';
