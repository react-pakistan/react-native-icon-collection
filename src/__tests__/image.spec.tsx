import React from 'react';
import renderer from 'react-test-renderer';
import { Image } from '../components/image';
import {
  imageLocalProps,
  imageUriProps,
} from '../../__stories__/story/image.story';

describe('Testing Image component', () => {
  it('matches Image snapshot, local asset', () => {
    const wrapper = renderer.create(
      <Image {...imageLocalProps} />,
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  it('matches Image snapshot, external asset', () => {
    const wrapper = renderer.create(
      <Image {...imageUriProps} />,
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
