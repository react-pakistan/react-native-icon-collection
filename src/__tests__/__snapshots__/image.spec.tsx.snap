// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Image component matches Image snapshot, external asset 1`] = `
<Image
  accessibilityLabel="Image"
  accessible={false}
  blurRadius={1}
  capInsets={
    Object {
      "bottom": 0,
      "left": 0,
      "right": 0,
      "top": 0,
    }
  }
  fadeDuration={300}
  height={100}
  loadingIndicatorSource={
    Array [
      Object {
        "testUri": "../../../assets/images/react-icon.png",
      },
      Object {
        "testUri": "../../../assets/images/react-icon.png",
      },
      Object {
        "testUri": "../../../assets/images/react-icon.png",
      },
    ]
  }
  onError={[Function]}
  onLayout={[Function]}
  onLoad={[Function]}
  onLoadEnd={[Function]}
  onLoadStart={[Function]}
  onPartialLoad={[Function]}
  onProgress={[Function]}
  progressiveRenderingEnabled={false}
  resizeMethod="auto"
  resizeMode="cover"
  source={
    Object {
      "uri": "https://facebook.github.io/react-native/img/tiny_logo.png",
    }
  }
  style={
    Array [
      Object {
        "height": 100,
        "width": 100,
      },
    ]
  }
  testID="testID"
  width={100}
/>
`;

exports[`Testing Image component matches Image snapshot, local asset 1`] = `
<Image
  accessibilityLabel="Image"
  accessible={false}
  blurRadius={1}
  capInsets={
    Object {
      "bottom": 0,
      "left": 0,
      "right": 0,
      "top": 0,
    }
  }
  fadeDuration={300}
  height={100}
  loadingIndicatorSource={
    Array [
      Object {
        "testUri": "../../../assets/images/react-icon.png",
      },
      Object {
        "testUri": "../../../assets/images/react-icon.png",
      },
      Object {
        "testUri": "../../../assets/images/react-icon.png",
      },
    ]
  }
  onError={[Function]}
  onLayout={[Function]}
  onLoad={[Function]}
  onLoadEnd={[Function]}
  onLoadStart={[Function]}
  onPartialLoad={[Function]}
  onProgress={[Function]}
  progressiveRenderingEnabled={false}
  resizeMethod="auto"
  resizeMode="cover"
  source={
    Object {
      "testUri": "../../../assets/images/react-icon.png",
    }
  }
  style={
    Array [
      Object {
        "height": 100,
        "width": 100,
      },
    ]
  }
  testID="testID"
  width={100}
/>
`;
